platform/ios/src/MGLMapView.mm

```diff
@@ -5994,9 +5994,14 @@ - (CGPoint)userLocationAnnotationViewCenter
         contentFrame = self.contentFrame;
     }
     CGPoint center = CGPointMake(CGRectGetMidX(contentFrame), CGRectGetMidY(contentFrame));
+    
+    if ([self.delegate respondsToSelector:@selector(mapViewUserLocationAnchorPoint:)])
+    {
+        center = [self.delegate mapViewUserLocationAnchorPoint:self];
+    }
 
-    // When tracking course, it’s more important to see the road ahead, so
-    // weight the user dot down towards the bottom.
+//     When tracking course, it’s more important to see the road ahead, so
+//     weight the user dot down towards the bottom.
     switch (self.userLocationVerticalAlignment) {
```

Comment:
This switch statement should only run if `-mapViewUserLocationAnchorPoint:` is unimplemented.